core/src/main/java/hudson/util/StreamCopyThread.java

```diff
@@ -63,8 +63,11 @@ public void run() {
                 // it doesn't make sense not to close InputStream that's already EOF-ed,
                 // so there's no 'closeIn' flag.
                 in.close();
-                if(closeOut)
+                if (closeOut) {
                     out.close();
+                } else {
+                    out.flush();
```

Comment:
See PR description for explanation.

Reply:
Would it make sense to inline this into a code comment to make it easier to understand what's going on here?